cmake/modules/BuildFIO.cmake

```diff
@@ -0,0 +1,28 @@
+include(CheckCXXSourceRuns)
+
+function(do_build_fio)
+  # we use an external project and copy the sources to bin directory to ensure
+  # that object files are built outside of the source tree.
+  include(ExternalProject)
+  ExternalProject_Add(fio_ext
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/fio
+    TMP_DIR ${CMAKE_BINARY_DIR}/fio_ext
+    STAMP_DIR ${CMAKE_BINARY_DIR}/fio_ext
+    BINARY_DIR ${CMAKE_BINARY_DIR}/fio_ext
+    CONFIGURE_COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/fio/configure
+    BUILD_COMMAND $(MAKE) fio EXTFLAGS=-Wno-format-truncation
+    COMMAND cp ${CMAKE_BINARY_DIR}/fio_ext/fio ${CMAKE_BINARY_DIR}/bin
+    INSTALL_COMMAND "true")
+ 
+  #ExternalProject_Get_Property(project_luajit install_dir)
+  # force fio make to be called on each time
+  ExternalProject_Add_Step(fio_ext forcebuild
```

Comment:
i'm guessing the forcebuild step is needed because `fio_ceph_objectstore` uses `FIO_INCLUDE_DIR` without expressing an actual dependency on this external project. i think it would be preferable to have both `Findfio.cmake` and `BuildFIO.cmake` add an imported target for `fio_ceph_objectstore` to use instead of using `FIO_INCLUDE_DIR` directly